python/ray/tune/tuner.py

```diff
@@ -251,8 +270,16 @@ def fit(self) -> ResultGrid:
             experiment_checkpoint_dir = ray.get(
                 self._remote_tuner.get_experiment_checkpoint_dir.remote()
             )
+            (
+                progress_reporter,
+                string_queue,
+            ) = self._prepare_remote_tuner_for_jupyter_progress_reporting()
             try:
-                return ray.get(self._remote_tuner.fit.remote())
+                return _get_remote_with_string_queue(
+                    self._remote_tuner.fit.remote(),
+                    progress_reporter,
+                    string_queue,
+                )
```

Comment:
Let's move the `ray.get` here to increase visibility on the actual return value.

Also, maybe rename to `_stream_client_output` and just immediately return if we have no string queue

```suggestion
                fit_future = self._remote_tuner.fit.remote()
                _stream_client_output(
                    fit_future,
                    progress_reporter,
                    string_queue,
                )
                return ray.get(fit_future)
```